sdk/storage/azblob/container/client_test.go

```diff
@@ -748,6 +748,45 @@ func (s *ContainerRecordedTestsSuite) TestContainerListBlobsInvalidDelimiter() {
 	}
 }
 
+func (s *ContainerRecordedTestsSuite) TestContainerListBlobsDelimiterPrefixVersion() {
+	_require := require.New(s.T())
+	testName := s.T().Name()
+	svcClient, err := testcommon.GetServiceClient(s.T(), testcommon.TestAccountDefault, nil)
+	_require.NoError(err)
+
+	containerName := testcommon.GenerateContainerName(testName)
+	containerClient := testcommon.CreateNewContainer(context.Background(), _require, containerName, svcClient)
+	defer testcommon.DeleteContainer(context.Background(), _require, containerClient)
+
+	blobNames := []string{"a/1", "a/2", "b/1", "blob"}
+	blobMap := testcommon.BlobListToMap(blobNames)
+	for _, name := range blobNames {
+		testcommon.CreateNewBlockBlob(context.Background(), _require, name, containerClient)
+	}
+
+	opts := container.ListBlobsHierarchyOptions{
```

Comment:
We also need to include the Delimiter.  See the .NET PR - https://github.com/Azure/azure-sdk-for-net/pull/30247/files#diff-cfbbd86d890e76d951dd6142c352ee932045fb0ce70ceb31c1d069f404c692c2R2929

Reply:
See the REST docs for more details about what the delimiter is - https://learn.microsoft.com/en-us/rest/api/storageservices/list-blobs?tabs=azure-ad#uri-parameters

Reply:
Delimiter for NewListBlobsHierarchyPager is the first parameter of the method (as seen in line  774). It is not listed as an option for ListBlobsHierarchyOptions. Is this something we should reconsider?

Reply:
Nope, this looks good, I missed that.